packages/cacti-plugin-consortium-static/src/main/typescript/plugin-consortium-static.ts

```diff
@@ -0,0 +1,503 @@
+      .sign(privateKey);
+    return signature;
+  }
+
+  async verifySignature(
+    signature: string,
+    msg: string,
+    pubKey: string,
+  ): Promise<boolean> {
+    const publicKey = await importSPKI(pubKey, "ES256K");
+    const { payload } = await compactVerify(signature, publicKey);
+
+    // Extract the original message
+    const payload1 = JSON.parse(payload.toString());
+    const signedData = new Uint8Array(payload1.data);
+    const decodedMessage = new TextDecoder().decode(signedData);
+    return decodedMessage === msg;
+  }
+
+  public async joinConsortium(managerApi: ConsortiumManagerApi, jwt: string) {
+    const fn = `${PluginConsortiumStatic.CLASS_NAME}#joinConsortium()`;
+    const nodeInfo = this.repo.getSelfData();
+    const treeHash =
+      this.repo.getPolicyTreeProof() + this.repo.getConfigsProof();
+
+    this.log.info(treeHash);
+    const msg = safeStableStringify({
+      identity: {
+        pubKey: this.options.keyPairPub,
+        memberId: nodeInfo.memberId,
+        proof: jwt, //token signed by the organization identified by nodeInfo.memberId
+      },
+      treeHash,
+      node: nodeInfo.node,
+      ledger: nodeInfo.ledgers,
+      pluginInstance: nodeInfo.pluginInstances,
+    }) as string;
+
+    const request: NewNodeRequestV1 = {
+      identity: {
+        pubKey: this.options.keyPairPub,
+        memberId: nodeInfo.memberId,
+        proof: jwt,
+      },
+      treeHash,
+      node: nodeInfo.node,
+      ledger: nodeInfo.ledgers,
+      pluginInstance: nodeInfo.pluginInstances,
+      signature: await this.sign(msg),
+    };
+
+    const response = await managerApi.addNodeToConsortiumV1(request);
+
+    if (!response) {
+      throw new InternalServerError(`${fn} response is falsy`);
+    }
+    if (!response.status) {
+      throw new InternalServerError(`${fn} response.status is falsy`);
+    }
+    const { status, data, statusText, config } = response;
+    if (response.status < 200 || response.status > 300) {
+      // We log the error here on the debug level so that later on we can inspect the contents
+      // of it in the logs if we need to. The reason that this is important is because we do not
+      // want to dump the full response onto our own error response that is going back to the caller
+      // due to that potentially being a security issue that we are exposing internal data via the
+      // error responses.
+      // With that said, we still need to make sure that we can determine the root cause of any
+      // issues after the fact and therefore we must save the error response details somewhere (the logs)
+      this.log.debug(
+        "ConsortiumApi non-2xx HTTP response:",
+        data,
+        status,
+        config,
+      );
+
+      // For the caller/client we just send back a generic error admitting that we somehow messed up:
+      const errorMessage = `${fn} BesuAPI error status: ${status}: ${statusText}`;
```

Comment:
@eduv09 Please double check if logging `BesuAPI` here is accurate. I'm guessing it should be something else!